### PR TITLE
Add M400 dummy command for Octoprint workaround

### PR DESCRIPTION
**Pull request by vegetablejedi**
- Octoprint calls M400 every printing job is ended.
- Fixed an issue where the M400 command was not implemented and disconnected each time.

### CODE REVIEW

File: g2core/marlin_compatibility.cpp [hunk 1]
```diff
@@ -406,14 +406,14 @@ stat_t marlin_report_version()
     char buffer[128];
     char *str = buffer;
 
-    str_concat(str, "ok FIRMWARE_NAME:Marlin g2core-");
+    str_concat(str, "ok FIRMWARE_NAME:Marlin g2core-vegetablejedi-");
```
Comment (justinclift) [inline g2core/marlin_compatibility.cpp:409]: Probably leave out the "vegetablejedi" for PR's to the main repo. :grin:

File: g2core/marlin_compatibility.cpp [hunk 2]
```diff
@@ -406,14 +406,14 @@ stat_t marlin_report_version()
...
     str_concat(str, G2CORE_FIRMWARE_BUILD_STRING);
     *str = 0;
     xio_writeline(buffer);
     str = buffer;
     *str = 0;
 
-    str_concat(str, " SOURCE_CODE_URL:https://github.com/synthetos/g2");
+    str_concat(str, " SOURCE_CODE_URL:https://github.com/synthetos/g2, https://github.com/vegetablejedi/g2");
```
Comment (justinclift) [inline g2core/marlin_compatibility.cpp:416]: As above. :smile:

File: g2core/gcode_parser.cpp [hunk 3]
```diff
@@ -840,6 +840,7 @@ static stat_t _parse_gcode_block(char *buf, char *active_comment)
 
                 case 115: SET_NON_MODAL (next_action, NEXT_ACTION_MARLIN_REPORT_VERSION);   // report version information
                 case 117: status = STAT_COMPLETE; break;  //SET_NON_MODAL (next_action, NEXT_ACTION_MARLIN_DISPLAY_ON_SCREEN);
+                case 400: status = STAT_COMPLETE; break; // Workaround for OctoPrint.
```
Comment (justinclift) [inline g2core/gcode_parser.cpp:843]: Sounds reasonable to me. :smile:

File: g2core/gcode_parser.cpp [hunk 4]
```diff
@@ -779,7 +779,7 @@ static stat_t _parse_gcode_block(char *buf, char *active_comment)
             break;
 
             case 'M':
-            switch((uint8_t)value) {
+            switch((uint16_t)value) {
```
Comment (justinclift) [inline g2core/gcode_parser.cpp:782]: Are you sure this is needed?

**Comment by justinclift** [comment]
The no-operation `M400` line seems like a reasonable approach to me. :smile:

The two new scripts (eg) `build-printrbot-simple-pro.sh` should probably be in a new PR though, as they're not strictly related to adding an M400 command. :smile:

File: g2core/marlin_compatibility.cpp (hunk 1 above)
Comment (vegetablejedi) [reply g2core/marlin_compatibility.cpp:409]: Yes. When I did a pull request, it was before I put the phrase, but after I committed it, the phrase went in like that. I suppose I had to work on the branch to pull requests. :-)

File: g2core/marlin_compatibility.cpp (hunk 2 above)
Comment (vegetablejedi) [reply g2core/marlin_compatibility.cpp:416]: Right.

File: g2core/gcode_parser.cpp (hunk 4 above)
Comment (vegetablejedi) [reply g2core/gcode_parser.cpp:782]: Yes. Because, 'uint8_t' cannot process '400'.

**Comment by vegetablejedi** [comment]
Should I fix the issue and pull-request again?

**Comment by vegetablejedi** [comment]
I'll rework!

File: g2core/gcode_parser.cpp (hunk 4 above)
Comment (justinclift) [reply g2core/gcode_parser.cpp:782]: Ahhh, of course.  Obvious now you point it out. :smile:

**Comment by justinclift** [comment]
> I'll rework!

With GitHub, you can just add more commits to the same branch you made the PR with (`edge` in this case).  When you next push commits to that branch on GitHub, they'll automatically be added to the same PR.

It's useful for fixing up stuff in PR's, adding more bits to them, and so on.

Since you're already made a new branch (`edge-M400-fix`), we might as well use that one. :grinning:

**Pull request closed without merging**